first_order_model/shrink_util.py

```diff
@@ -951,82 +1014,83 @@ def compute_deletion(layer_graph,
     for after_layer in layer_graph[layer].after:
         follow(following_layers, layer_graph, layer_graph[after_layer])
 
-    # If you delete a part of the previous one, not just all of it then you cannot just say these n are to be deleted
-    # So for each layer you need to figure out what you are deleting from it, then for the next layer figure out what is deleted
-
     deletions = {}
     if isinstance(custom, list):
-        deletions[layer] = custom
+        deletions[layer] = copy.copy(custom)
     elif isinstance(custom, int):
         # The sorting magic happens here
         # Find the first convolution
         if sort:
-            if layer_graph[layer].type == 'bn':
-                breakpoint()
+            assert layer_graph[layer].type != 'bn', "We cant sort batchnorm layers"
+
+            # Get the layer which accepts this ones output as input.
             depthwise_skipped_following_layers = []
             for after_layer in layer_graph[layer].after:
                 get_following_layers_skip_depthwise(depthwise_skipped_following_layers, layer_graph, layer_graph[after_layer])
 
             first_conv = get_first_conv(layer_graph, depthwise_skipped_following_layers)
```

Comment:
The comments here are really not helpful for understanding what's going on. You need a more detailed step by step breakdown that suggests something like. Consider deleting this layer, see if there are skip connections that use it, if so, keep tracking slices until the next convolution is reached that breaks this dependence, compute importances of each filter throughout this path, delete the least important. Point to each of these steps in line comments also in addition to a full block comment

Reply:
I think I have made this a lot better now.